### PR TITLE
Add docker readme to general documentation

### PR DESCRIPTION
**Pull request by GernotMaier**
Add [readme](docker/README.md) file to general documentation.

- add the myst-parser package to allow to use markdown in sphinx (see the [sphinx web page](https://www.sphinx-doc.org/en/master/usage/markdown.html) for the recommended way)
- updated docker readme file (minor text updates)
- integrate readme into general documentation

### CODE REVIEW

**Comment by GernotMaier** [comment]
See screenshot, in case you don't want to build it on your own.

Screenshot 2024-05-22 at 08 56 52

File: docs/source/docker_files.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+```{include} ../../docker/README.md
+```
```
Comment (VictorBarbosaMartins) [inline docs/source/docker_files.md:2]: This is a kind of duplication I think. An entire file only to call another one. Couldn't we include the README directly in the source index and not use md for the moment? This file is the one thing that would justify adding the package but the file itself is not necessary.

**Review by VictorBarbosaMartins — COMMENTED** [review]
Hi @GernotMaier, I think we need to discuss this before proceeding. I don't think this is necessary or justified at the moment (see comments below)

File: docker/README.md [hunk 2]
```diff
@@ -1,36 +1,52 @@
...
+## Simtools package registry
+
+Pre-built container images are available through the [simtools package registry](https://github.com/orgs/gammasim/packages?repo_name=simtools).
+
+Follow the [instruction](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry) (`docker login`) to authenticate with the GitHub package registry before pulling the first image.
+
+Note: if the docker image already exists in your system, this same image will be used to run the container. This might especially be an issue when using the `latest` tag, which might not be the latest version available in the registry. To force a pull of the latest image, use the `--pull` option, e.g.:
+
+```bash
+docker run --pull always --rm -it ghcr.io/gammasim/simtools-prod:latest bash
```
[the remark this answers is not in the document]
Comment (GernotMaier) [reply docker/README.md:25]: Not sure - I do think that almost everyone (outside DESY) is using docker. Also is docker a synonym for container engines, so I think this is clear enough (and to make it easy: add an alias on your local machine from podman to docker)

[the remark this answers is not in the document]
Comment (VictorBarbosaMartins) [reply docker/README.md:25]: If you want to keep the docker commands, then I suggest adding that docker as one of the possible softwares among with the already mentioned podman and apptainer (line 5) and mention that the commands here are provided for docker but they are similar among all the softwares.

File: docs/source/docker_files.md (hunk 1 above)
Comment (VictorBarbosaMartins) [reply docs/source/docker_files.md:2]: Ok for including the file then, but using md in a single file does not justify adding the package to support md in the documentation. Unless we start using md more frequently, adding the package is not well motivated for me.

File: docker/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (GernotMaier) [reply docker/README.md:25]: This is the most basic thing one learns when installing any of the container engines, so I don't think this is really necessary.

**Review by VictorBarbosaMartins — APPROVED** [review]
Thank you for these changes and the discussions, I think we can move forward here.

**Comment by GernotMaier** [comment]
Thank you!

**Pull request merged**